[JSC] Do not use Variant for each of TypeDefinition kind

Constellation opened the pull request:
#### dc1ef6a78206884b6ef022cc4e71497e4cb9e8d2
<pre>
[JSC] Do not use Variant for each of TypeDefinition kind
<a href="https://bugs.webkit.org/show_bug.cgi?id=297407">https://bugs.webkit.org/show_bug.cgi?id=297407</a>
<a href="https://rdar.apple.com/158334942">rdar://158334942</a>

Reviewed by Justin Michaud and Keith Miller.

FunctionSignature etc. must be just a derived type of TypeDefinition.
There is no reason having Variant and making code much more unsafe and
complicated. This patch just makes FunctionSignature, StructType etc. as
a derived type of TypeDefinition. We can do further more clean up in
various places. But this patch starts doing this clean up in WasmTypeDefinition.

* Source/JavaScriptCore/SaferCPPExpectations/UncountedCallArgsCheckerExpectations:
* Source/JavaScriptCore/runtime/IteratorOperations.h:
(JSC::forEachInFastArray):
(JSC::forEachInIterationRecord):
(JSC::forEachInIterable):
(JSC::forEachInIteratorProtocol):
* Source/JavaScriptCore/wasm/WasmCallingConvention.h:
(JSC::Wasm::WasmCallingConvention::callInformationFor const):
(JSC::Wasm::JSCallingConvention::callInformationFor const):
* Source/JavaScriptCore/wasm/WasmOMGIRGenerator.cpp:
(JSC::Wasm::OMGIRGenerator::addArguments):
* Source/JavaScriptCore/wasm/WasmOperations.cpp:
(JSC::Wasm::JSC_DEFINE_NOEXCEPT_JIT_OPERATION):
* Source/JavaScriptCore/wasm/WasmTypeDefinition.cpp:
(JSC::Wasm::TypeDefinition::dump const):
(JSC::Wasm::FunctionSignature::FunctionSignature):
(JSC::Wasm::StructType::StructType):
(JSC::Wasm::ArrayType::ArrayType):
(JSC::Wasm::RecursionGroup::RecursionGroup):
(JSC::Wasm::Projection::Projection):
(JSC::Wasm::Projection::dump const):
(JSC::Wasm::Subtype::Subtype):
(JSC::Wasm::Subtype::cleanup):
(JSC::Wasm::RecursionGroup::cleanup):
(JSC::Wasm::computeStructTypeHash):
(JSC::Wasm::computeRecursionGroupHash):
(JSC::Wasm::computeProjectionHash):
(JSC::Wasm::computeSubtypeHash):
(JSC::Wasm::TypeDefinition::hash const):
(JSC::Wasm::FunctionSignature::tryCreate):
(JSC::Wasm::StructType::tryCreate):
(JSC::Wasm::ArrayType::tryCreate):
(JSC::Wasm::RecursionGroup::tryCreate):
(JSC::Wasm::Projection::tryCreate):
(JSC::Wasm::Subtype::tryCreate):
(JSC::Wasm::TypeDefinition::substitute):
(JSC::Wasm::substituteParent):
(JSC::Wasm::TypeDefinition::replacePlaceholders const):
(JSC::Wasm::TypeDefinition::unrollSlow const):
(JSC::Wasm::RTT::tryCreate):
(JSC::Wasm::TypeInformation::signatureForJSException):
(JSC::Wasm::FunctionParameterTypes::translate):
(JSC::Wasm::StructParameterTypes::hash):
(JSC::Wasm::StructParameterTypes::equal):
(JSC::Wasm::StructParameterTypes::translate):
(JSC::Wasm::ArrayParameterTypes::translate):
(JSC::Wasm::RecursionGroupParameterTypes::hash):
(JSC::Wasm::RecursionGroupParameterTypes::translate):
(JSC::Wasm::ProjectionParameterTypes::hash):
(JSC::Wasm::ProjectionParameterTypes::equal):
(JSC::Wasm::ProjectionParameterTypes::translate):
(JSC::Wasm::SubtypeParameterTypes::hash):
(JSC::Wasm::SubtypeParameterTypes::translate):
(JSC::Wasm::TypeInformation::TypeInformation):
(JSC::Wasm::TypeInformation::typeDefinitionForFunction):
(JSC::Wasm::TypeInformation::typeDefinitionForStruct):
(JSC::Wasm::TypeInformation::typeDefinitionForArray):
(JSC::Wasm::TypeInformation::typeDefinitionForRecursionGroup):
(JSC::Wasm::TypeInformation::typeDefinitionForProjection):
(JSC::Wasm::TypeInformation::typeDefinitionForSubtype):
(JSC::Wasm::TypeInformation::getPlaceholderProjection):
(JSC::Wasm::TypeInformation::canonicalRTTForType):
(JSC::Wasm::TypeDefinition::tryCreateFunctionSignature): Deleted.
(JSC::Wasm::TypeDefinition::tryCreateStructType): Deleted.
(JSC::Wasm::TypeDefinition::tryCreateArrayType): Deleted.
(JSC::Wasm::TypeDefinition::tryCreateRecursionGroup): Deleted.
(JSC::Wasm::TypeDefinition::tryCreateProjection): Deleted.
(JSC::Wasm::TypeDefinition::tryCreateSubtype): Deleted.
(JSC::Wasm::RTT::tryCreateRTT): Deleted.
* Source/JavaScriptCore/wasm/WasmTypeDefinition.h:
(JSC::Wasm::TypeDefinition::is const):
(JSC::Wasm::TypeDefinition::as):
(JSC::Wasm::TypeDefinition::as const):
(JSC::Wasm::TypeDefinition::operator== const):
(JSC::Wasm::TypeDefinition::unroll const):
(JSC::Wasm::TypeDefinition::unownedIndex const):
(JSC::Wasm::TypeDefinition::TypeDefinition):
(JSC::Wasm:: const):
(JSC::Wasm::FunctionSignature::argumentCount const): Deleted.
(JSC::Wasm::FunctionSignature::returnCount const): Deleted.
(JSC::Wasm::FunctionSignature::hasRecursiveReference const): Deleted.
(JSC::Wasm::FunctionSignature::setHasRecursiveReference): Deleted.
(JSC::Wasm::FunctionSignature::returnType const): Deleted.
(JSC::Wasm::FunctionSignature::returnsVoid const): Deleted.
(JSC::Wasm::FunctionSignature::argumentType const): Deleted.
(JSC::Wasm::FunctionSignature::argumentsOrResultsIncludeI64 const): Deleted.
(JSC::Wasm::FunctionSignature::setArgumentsOrResultsIncludeI64): Deleted.
(JSC::Wasm::FunctionSignature::argumentsOrResultsIncludeV128 const): Deleted.
(JSC::Wasm::FunctionSignature::setArgumentsOrResultsIncludeV128): Deleted.
(JSC::Wasm::FunctionSignature::argumentsOrResultsIncludeExnref const): Deleted.
(JSC::Wasm::FunctionSignature::setArgumentsOrResultsIncludeExnref): Deleted.
(JSC::Wasm::FunctionSignature::numVectors const): Deleted.
(JSC::Wasm::FunctionSignature::numReturnVectors const): Deleted.
(JSC::Wasm::FunctionSignature::hasReturnVector const): Deleted.
(JSC::Wasm::FunctionSignature::operator== const): Deleted.
(JSC::Wasm::FunctionSignature::getReturnType): Deleted.
(JSC::Wasm::FunctionSignature::getArgumentType): Deleted.
(JSC::Wasm::FunctionSignature::storage): Deleted.
(JSC::Wasm::FunctionSignature::storage const): Deleted.
(JSC::Wasm::StructType::fieldCount const): Deleted.
(JSC::Wasm::StructType::field const): Deleted.
(JSC::Wasm::StructType::hasRefFieldTypes const): Deleted.
(JSC::Wasm::StructType::hasRecursiveReference const): Deleted.
(JSC::Wasm::StructType::setHasRecursiveReference): Deleted.
(JSC::Wasm::StructType::getField): Deleted.
(JSC::Wasm::StructType::storage): Deleted.
(JSC::Wasm::StructType::storage const): Deleted.
(JSC::Wasm::StructType::offsetOfFieldInPayload const): Deleted.
(JSC::Wasm::StructType::instancePayloadSize const): Deleted.
(JSC::Wasm::StructType::fieldOffsetFromInstancePayload): Deleted.
(JSC::Wasm::ArrayType::ArrayType): Deleted.
(JSC::Wasm::ArrayType::elementType const): Deleted.
(JSC::Wasm::ArrayType::hasRecursiveReference const): Deleted.
(JSC::Wasm::ArrayType::setHasRecursiveReference): Deleted.
(JSC::Wasm::ArrayType::getElementType): Deleted.
(JSC::Wasm::ArrayType::storage): Deleted.
(JSC::Wasm::ArrayType::storage const): Deleted.
(JSC::Wasm::RecursionGroup::RecursionGroup): Deleted.
(JSC::Wasm::RecursionGroup::typeCount const): Deleted.
(JSC::Wasm::RecursionGroup::type const): Deleted.
(JSC::Wasm::RecursionGroup::getType): Deleted.
(JSC::Wasm::RecursionGroup::storage): Deleted.
(JSC::Wasm::RecursionGroup::storage const): Deleted.
(JSC::Wasm::Projection::Projection): Deleted.
(JSC::Wasm::Projection::recursionGroup const): Deleted.
(JSC::Wasm::Projection::index const): Deleted.
(JSC::Wasm::Projection::getRecursionGroup): Deleted.
(JSC::Wasm::Projection::getIndex): Deleted.
(JSC::Wasm::Projection::storage): Deleted.
(JSC::Wasm::Projection::storage const): Deleted.
(JSC::Wasm::Projection::isPlaceholder const): Deleted.
(JSC::Wasm::Subtype::Subtype): Deleted.
(JSC::Wasm::Subtype::supertypeCount const): Deleted.
(JSC::Wasm::Subtype::isFinal const): Deleted.
(JSC::Wasm::Subtype::firstSuperType const): Deleted.
(JSC::Wasm::Subtype::superType const): Deleted.
(JSC::Wasm::Subtype::underlyingType const): Deleted.
(JSC::Wasm::Subtype::getSuperType): Deleted.
(JSC::Wasm::Subtype::getUnderlyingType): Deleted.
(JSC::Wasm::Subtype::storage): Deleted.
(JSC::Wasm::Subtype::storage const): Deleted.
(JSC::Wasm::TypeDefinition::payload): Deleted.
(JSC::Wasm::TypeDefinition::allocatedFunctionSize): Deleted.
(JSC::Wasm::TypeDefinition::allocatedStructSize): Deleted.
(JSC::Wasm::TypeDefinition::allocatedArraySize): Deleted.
(JSC::Wasm::TypeDefinition::allocatedRecursionGroupSize): Deleted.
(JSC::Wasm::TypeDefinition::allocatedProjectionSize): Deleted.
(JSC::Wasm::TypeDefinition::allocatedSubtypeSize): Deleted.
* Source/JavaScriptCore/wasm/js/JSWebAssemblyException.cpp:
(JSC::JSWebAssemblyException::visitChildrenImpl):
(JSC::JSWebAssemblyException::getArg const):
* Source/JavaScriptCore/wasm/js/JSWebAssemblyStruct.cpp:
(JSC::JSWebAssemblyStruct::tryCreate):
* Source/JavaScriptCore/wasm/js/WebAssemblyCompileOptions.cpp:
(JSC::WebAssemblyCompileOptions::validateImportForBuiltinSetNames const):
* Source/JavaScriptCore/wasm/js/WebAssemblyExceptionConstructor.cpp:
(JSC::JSC_DEFINE_HOST_FUNCTION):
* Source/JavaScriptCore/wasm/js/WebAssemblyModuleConstructor.cpp:
(JSC::createTypeReflectionObject):
* Source/JavaScriptCore/wasm/js/WebAssemblyWrapperFunction.cpp:
(JSC::WebAssemblyWrapperFunction::create):

Canonical link: <a href="https://commits.webkit.org/298730@main">https://commits.webkit.org/298730@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/5904e602803a5542adbc1d9cfb1387cbb3117612

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [❌ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/116502 "16 style errors") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/36163 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/26715 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/122559 "Built successfully") | [  ~~🛠 win~~](https://ews-build.webkit.org/#/builders/59/builds/67065 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | ⏳ 🛠 ios-apple 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/36857 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/44751 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/88470 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/59/builds/67065 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | ⏳ 🛠 mac-apple 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/119451 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/29392 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/104509 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/68915 "Passed tests") | | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/133/builds/28459 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/135/builds/22617 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/66240 "Built successfully") | | 
| [  ~~🛠 🧪 jsc~~](https://ews-build.webkit.org/#/builders/20/builds/108612 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/98767 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/22776 "Passed tests") | [  ~~🛠 gtk~~](https://ews-build.webkit.org/#/builders/2/builds/125708 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/115030 "Built successfully and passed tests") | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/43396 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/32586 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/2/builds/125708 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/43761 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/100718 "Passed tests") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/2/builds/125708 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/42267 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/20177 "Passed tests") | [  ~~🛠 playstation~~](https://ews-build.webkit.org/#/builders/134/builds/39350 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/18598 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/43284 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/48875 "Built successfully") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/143728 "Built successfully") | | 
| | [  ~~🛠 tv-sim~~](https://ews-build.webkit.org/#/builders/125/builds/42750 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/37014 "Passed tests") | | 
| | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/129/builds/46090 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/44455 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/259df1cf33da0a016de56e0f50dfbe967ef61ee5. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/49394

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/37643355ce2a734e9829eb8422dc5dccbe2d1fcf. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/49394

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/37fceb5ef7806d996be75d327d987a58c4103c37. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/49394

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/ed878d38c8bf7bae4c04c586c8e2db8fa5bc19c2. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/49394

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/ae4299706ea9be5847dfb6418e4a045abb43741d. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/49394

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/5904e602803a5542adbc1d9cfb1387cbb3117612. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/49394

Constellation committed:
[JSC] Do not use Variant for each of TypeDefinition kind
https://bugs.webkit.org/show_bug.cgi?id=297407
rdar://158334942

Reviewed by Justin Michaud and Keith Miller.

FunctionSignature etc. must be just a derived type of TypeDefinition.
There is no reason having Variant and making code much more unsafe and
complicated. This patch just makes FunctionSignature, StructType etc. as
a derived type of TypeDefinition. We can do further more clean up in
various places. But this patch starts doing this clean up in WasmTypeDefinition.

* Source/JavaScriptCore/SaferCPPExpectations/UncountedCallArgsCheckerExpectations:
* Source/JavaScriptCore/runtime/IteratorOperations.h:
(JSC::forEachInFastArray):
(JSC::forEachInIterationRecord):
(JSC::forEachInIterable):
(JSC::forEachInIteratorProtocol):
* Source/JavaScriptCore/wasm/WasmCallingConvention.h:
(JSC::Wasm::WasmCallingConvention::callInformationFor const):
(JSC::Wasm::JSCallingConvention::callInformationFor const):
* Source/JavaScriptCore/wasm/WasmOMGIRGenerator.cpp:
(JSC::Wasm::OMGIRGenerator::addArguments):
* Source/JavaScriptCore/wasm/WasmOperations.cpp:
(JSC::Wasm::JSC_DEFINE_NOEXCEPT_JIT_OPERATION):
* Source/JavaScriptCore/wasm/WasmTypeDefinition.cpp:
(JSC::Wasm::TypeDefinition::dump const):
(JSC::Wasm::FunctionSignature::FunctionSignature):
(JSC::Wasm::StructType::StructType):
(JSC::Wasm::ArrayType::ArrayType):
(JSC::Wasm::RecursionGroup::RecursionGroup):
(JSC::Wasm::Projection::Projection):
(JSC::Wasm::Projection::dump const):
(JSC::Wasm::Subtype::Subtype):
(JSC::Wasm::Subtype::cleanup):
(JSC::Wasm::RecursionGroup::cleanup):
(JSC::Wasm::computeStructTypeHash):
(JSC::Wasm::computeRecursionGroupHash):
(JSC::Wasm::computeProjectionHash):
(JSC::Wasm::computeSubtypeHash):
(JSC::Wasm::TypeDefinition::hash const):
(JSC::Wasm::FunctionSignature::tryCreate):
(JSC::Wasm::StructType::tryCreate):
(JSC::Wasm::ArrayType::tryCreate):
(JSC::Wasm::RecursionGroup::tryCreate):
(JSC::Wasm::Projection::tryCreate):
(JSC::Wasm::Subtype::tryCreate):
(JSC::Wasm::TypeDefinition::substitute):
(JSC::Wasm::substituteParent):
(JSC::Wasm::TypeDefinition::replacePlaceholders const):
(JSC::Wasm::TypeDefinition::unrollSlow const):
(JSC::Wasm::RTT::tryCreate):
(JSC::Wasm::TypeInformation::signatureForJSException):
(JSC::Wasm::FunctionParameterTypes::translate):
(JSC::Wasm::StructParameterTypes::hash):
(JSC::Wasm::StructParameterTypes::equal):
(JSC::Wasm::StructParameterTypes::translate):
(JSC::Wasm::ArrayParameterTypes::translate):
(JSC::Wasm::RecursionGroupParameterTypes::hash):
(JSC::Wasm::RecursionGroupParameterTypes::translate):
(JSC::Wasm::ProjectionParameterTypes::hash):
(JSC::Wasm::ProjectionParameterTypes::equal):
(JSC::Wasm::ProjectionParameterTypes::translate):
(JSC::Wasm::SubtypeParameterTypes::hash):
(JSC::Wasm::SubtypeParameterTypes::translate):
(JSC::Wasm::TypeInformation::TypeInformation):
(JSC::Wasm::TypeInformation::typeDefinitionForFunction):
(JSC::Wasm::TypeInformation::typeDefinitionForStruct):
(JSC::Wasm::TypeInformation::typeDefinitionForArray):
(JSC::Wasm::TypeInformation::typeDefinitionForRecursionGroup):
(JSC::Wasm::TypeInformation::typeDefinitionForProjection):
(JSC::Wasm::TypeInformation::typeDefinitionForSubtype):
(JSC::Wasm::TypeInformation::getPlaceholderProjection):
(JSC::Wasm::TypeInformation::canonicalRTTForType):
(JSC::Wasm::TypeDefinition::tryCreateFunctionSignature): Deleted.
(JSC::Wasm::TypeDefinition::tryCreateStructType): Deleted.
(JSC::Wasm::TypeDefinition::tryCreateArrayType): Deleted.
(JSC::Wasm::TypeDefinition::tryCreateRecursionGroup): Deleted.
(JSC::Wasm::TypeDefinition::tryCreateProjection): Deleted.
(JSC::Wasm::TypeDefinition::tryCreateSubtype): Deleted.
(JSC::Wasm::RTT::tryCreateRTT): Deleted.
* Source/JavaScriptCore/wasm/WasmTypeDefinition.h:
(JSC::Wasm::TypeDefinition::is const):
(JSC::Wasm::TypeDefinition::as):
(JSC::Wasm::TypeDefinition::as const):
(JSC::Wasm::TypeDefinition::operator== const):
(JSC::Wasm::TypeDefinition::unroll const):
(JSC::Wasm::TypeDefinition::unownedIndex const):
(JSC::Wasm::TypeDefinition::TypeDefinition):
(JSC::Wasm:: const):
(JSC::Wasm::FunctionSignature::argumentCount const): Deleted.
(JSC::Wasm::FunctionSignature::returnCount const): Deleted.
(JSC::Wasm::FunctionSignature::hasRecursiveReference const): Deleted.
(JSC::Wasm::FunctionSignature::setHasRecursiveReference): Deleted.
(JSC::Wasm::FunctionSignature::returnType const): Deleted.
(JSC::Wasm::FunctionSignature::returnsVoid const): Deleted.
(JSC::Wasm::FunctionSignature::argumentType const): Deleted.
(JSC::Wasm::FunctionSignature::argumentsOrResultsIncludeI64 const): Deleted.
(JSC::Wasm::FunctionSignature::setArgumentsOrResultsIncludeI64): Deleted.
(JSC::Wasm::FunctionSignature::argumentsOrResultsIncludeV128 const): Deleted.
(JSC::Wasm::FunctionSignature::setArgumentsOrResultsIncludeV128): Deleted.
(JSC::Wasm::FunctionSignature::argumentsOrResultsIncludeExnref const): Deleted.
(JSC::Wasm::FunctionSignature::setArgumentsOrResultsIncludeExnref): Deleted.
(JSC::Wasm::FunctionSignature::numVectors const): Deleted.
(JSC::Wasm::FunctionSignature::numReturnVectors const): Deleted.
(JSC::Wasm::FunctionSignature::hasReturnVector const): Deleted.
(JSC::Wasm::FunctionSignature::operator== const): Deleted.
(JSC::Wasm::FunctionSignature::getReturnType): Deleted.
(JSC::Wasm::FunctionSignature::getArgumentType): Deleted.
(JSC::Wasm::FunctionSignature::storage): Deleted.
(JSC::Wasm::FunctionSignature::storage const): Deleted.
(JSC::Wasm::StructType::fieldCount const): Deleted.
(JSC::Wasm::StructType::field const): Deleted.
(JSC::Wasm::StructType::hasRefFieldTypes const): Deleted.
(JSC::Wasm::StructType::hasRecursiveReference const): Deleted.
(JSC::Wasm::StructType::setHasRecursiveReference): Deleted.
(JSC::Wasm::StructType::getField): Deleted.
(JSC::Wasm::StructType::storage): Deleted.
(JSC::Wasm::StructType::storage const): Deleted.
(JSC::Wasm::StructType::offsetOfFieldInPayload const): Deleted.
(JSC::Wasm::StructType::instancePayloadSize const): Deleted.
(JSC::Wasm::StructType::fieldOffsetFromInstancePayload): Deleted.
(JSC::Wasm::ArrayType::ArrayType): Deleted.
(JSC::Wasm::ArrayType::elementType const): Deleted.
(JSC::Wasm::ArrayType::hasRecursiveReference const): Deleted.
(JSC::Wasm::ArrayType::setHasRecursiveReference): Deleted.
(JSC::Wasm::ArrayType::getElementType): Deleted.
(JSC::Wasm::ArrayType::storage): Deleted.
(JSC::Wasm::ArrayType::storage const): Deleted.
(JSC::Wasm::RecursionGroup::RecursionGroup): Deleted.
(JSC::Wasm::RecursionGroup::typeCount const): Deleted.
(JSC::Wasm::RecursionGroup::type const): Deleted.
(JSC::Wasm::RecursionGroup::getType): Deleted.
(JSC::Wasm::RecursionGroup::storage): Deleted.
(JSC::Wasm::RecursionGroup::storage const): Deleted.
(JSC::Wasm::Projection::Projection): Deleted.
(JSC::Wasm::Projection::recursionGroup const): Deleted.
(JSC::Wasm::Projection::index const): Deleted.
(JSC::Wasm::Projection::getRecursionGroup): Deleted.
(JSC::Wasm::Projection::getIndex): Deleted.
(JSC::Wasm::Projection::storage): Deleted.
(JSC::Wasm::Projection::storage const): Deleted.
(JSC::Wasm::Projection::isPlaceholder const): Deleted.
(JSC::Wasm::Subtype::Subtype): Deleted.
(JSC::Wasm::Subtype::supertypeCount const): Deleted.
(JSC::Wasm::Subtype::isFinal const): Deleted.
(JSC::Wasm::Subtype::firstSuperType const): Deleted.
(JSC::Wasm::Subtype::superType const): Deleted.
(JSC::Wasm::Subtype::underlyingType const): Deleted.
(JSC::Wasm::Subtype::getSuperType): Deleted.
(JSC::Wasm::Subtype::getUnderlyingType): Deleted.
(JSC::Wasm::Subtype::storage): Deleted.
(JSC::Wasm::Subtype::storage const): Deleted.
(JSC::Wasm::TypeDefinition::payload): Deleted.
(JSC::Wasm::TypeDefinition::allocatedFunctionSize): Deleted.
(JSC::Wasm::TypeDefinition::allocatedStructSize): Deleted.
(JSC::Wasm::TypeDefinition::allocatedArraySize): Deleted.
(JSC::Wasm::TypeDefinition::allocatedRecursionGroupSize): Deleted.
(JSC::Wasm::TypeDefinition::allocatedProjectionSize): Deleted.
(JSC::Wasm::TypeDefinition::allocatedSubtypeSize): Deleted.
* Source/JavaScriptCore/wasm/js/JSWebAssemblyException.cpp:
(JSC::JSWebAssemblyException::visitChildrenImpl):
(JSC::JSWebAssemblyException::getArg const):
* Source/JavaScriptCore/wasm/js/JSWebAssemblyStruct.cpp:
(JSC::JSWebAssemblyStruct::tryCreate):
* Source/JavaScriptCore/wasm/js/WebAssemblyCompileOptions.cpp:
(JSC::WebAssemblyCompileOptions::validateImportForBuiltinSetNames const):
* Source/JavaScriptCore/wasm/js/WebAssemblyExceptionConstructor.cpp:
(JSC::JSC_DEFINE_HOST_FUNCTION):
* Source/JavaScriptCore/wasm/js/WebAssemblyModuleConstructor.cpp:
(JSC::createTypeReflectionObject):
* Source/JavaScriptCore/wasm/js/WebAssemblyWrapperFunction.cpp:
(JSC::WebAssemblyWrapperFunction::create):

Canonical link: https://commits.webkit.org/298730@main
